crates/debug/src/transform/expression.rs

```diff
@@ -727,7 +877,121 @@ mod tests {
+            DW_OP_WASM_location,
+            0x0,
+            1,
+            DW_OP_and,
+            DW_OP_bra,
+            7,
+            0, // --> pointer
+            DW_OP_swap,
+            DW_OP_shr,
+            DW_OP_skip,
+            2,
+            0, // --> done
+            // pointer:
+            DW_OP_plus,
+            DW_OP_deref,
+            // done:
+            DW_OP_stack_value
+        );
+        let ce = compile_expression(&e, DWARF_ENCODING, None)
+            .expect("non-error")
+            .expect("expression");
+        assert_eq!(
+            ce,
+            CompiledExpression {
+                parts: vec![
+                    CompiledExpressionPart::Code(vec![49, 18]),
+                    CompiledExpressionPart::Local {
+                        label: val1,
+                        trailing: false
+                    },
+                    CompiledExpressionPart::Code(vec![26]),
+                    CompiledExpressionPart::Jump {
+                        conditionally: true
+                    },
+                    CompiledExpressionPart::LandingPad { original_pos: 9 }, // capture from
+                    CompiledExpressionPart::Code(vec![22, 37]),
+                    CompiledExpressionPart::Jump {
+                        conditionally: false
+                    },
+                    CompiledExpressionPart::LandingPad { original_pos: 14 }, // capture from
+                    CompiledExpressionPart::Code(vec![34]),
+                    CompiledExpressionPart::Deref,
+                    CompiledExpressionPart::LandingPad { original_pos: 15 }, // capture to
+                    CompiledExpressionPart::Code(vec![6, 159])
+                ],
+                need_deref: false,
+                jump_arcs: {
+                    let mut m = HashMap::new();
+                    m.insert(9, 16);
+                    m.insert(14, 16);
+                    m
+                }
+            }
+        );
+
+        let e = expression!(
+            DW_OP_lit1,
+            DW_OP_dup,
+            DW_OP_bra,
+            2,
+            0, // --> target
+            DW_OP_deref,
+            DW_OP_lit0,
+            // target:
+            DW_OP_stack_value
+        );
+        let ce = compile_expression(&e, DWARF_ENCODING, None)
+            .expect("non-error")
+            .expect("expression");
+        assert_eq!(
+            ce,
+            CompiledExpression {
+                parts: vec![
+                    CompiledExpressionPart::Code(vec![49, 18]),
+                    CompiledExpressionPart::Jump {
+                        conditionally: true
+                    },
+                    CompiledExpressionPart::LandingPad { original_pos: 5 }, // capture from
+                    CompiledExpressionPart::Deref,
+                    CompiledExpressionPart::LandingPad { original_pos: 5 }, // capture to
+                    CompiledExpressionPart::Code(vec![6, 48, 159])
```

Comment:
I think this and above test cases are incorrect. Can you verify that?

Reply:
The parsing tests pass
```
test transform::expression::tests::test_debug_parse_expressions ... ok
```
The tests themselves are not intended to make sense, if that is what you mean. The example in the PR description is an actual location expression with its translation.

Reply:
The `original_pos: 5` appearing twice is probably which is confusing. It confused me too. But actually it makes sense. The jump goes from 5 to 7. The `LandingPad` before `Code` just orients it and tells that there are potential jump targets in `Code` (pre-translation) pos 6: `DW_OP_lit0`, pos 7: `DW_OP_stack_value`.